File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/backend/connected/aas/submodelelement/property/file/ConnectedFile.java
package org.eclipse.basyx.aas.backend.connected.aas.submodelelement.property.file;

import org.eclipse.basyx.aas.api.metamodel.aas.submodelelement.property.file.IFile;
import org.eclipse.basyx.aas.backend.connected.aas.submodelelement.ConnectedDataElement;
import org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.submodelelement.property.SingleProperty;
import org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.submodelelement.property.blob.Blob;
import org.eclipse.basyx.vab.core.proxy.VABElementProxy;
/**
 * "Connected" implementation of IFile
 * @author rajashek
 *
 */
public class ConnectedFile extends ConnectedDataElement implements IFile {
	public ConnectedFile(VABElementProxy proxy) {
		super(proxy);		
	}
	
	@Override
	public void setValue(String value) {
		getProxy().setModelPropertyValue(SingleProperty.VALUE, value);
		
	}

	@Override
	public String getValue() {
		return (String) getProxy().getModelPropertyValue(SingleProperty.VALUE);
	}

	@Override
	public void setMimeType(String mimeType) {
		getProxy().setModelPropertyValue(Blob.MIMETYPE, mimeType);
		
	}

	@Override
	public String getMimeType() {
		return (String) getProxy().getModelPropertyValue(Blob.MIMETYPE);
	}

}

File: sdks/c++/basys.sdk.cc/src/aas/backend/connected/aas/submodelelement/property/file/ConnectedFile.cpp
/*
 * ConnectedFile.cpp
 *
 *      Author: wendel
 */

#include "ConnectedFile.h"
#include "aas/submodelelement/property/IProperty.h"
#include "aas/submodelelement/property/blob/IBlob.h"

namespace basyx {
namespace aas {
namespace backend {
namespace connected {

ConnectedFile::ConnectedFile(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  ConnectedDataElement(proxy)
{}

void ConnectedFile::setValue(const std::string & value)
{
  this->setProxyValue(submodelelement::property::PropertyPaths::VALUE, value);
}

std::string ConnectedFile::getValue() const
{
  return this->getProxyValue(submodelelement::property::PropertyPaths::VALUE);
}

void ConnectedFile::setMimeType(const std::string & mimeType)
{
  this->setProxyValue(submodelelement::BlobPath::MIMETYPE, mimeType);
}

std::string ConnectedFile::getMimeType() const
{
  return this->getProxyValue(submodelelement::BlobPath::MIMETYPE);
}


}
}
}
}
